monkey/infection_monkey/master/automated_master.py

```diff
@@ -195,11 +195,8 @@ def _run_plugins(self, plugin: List[Any], plugin_type: str, callback: Callable[[
         logger.info(f"Running {plugin_type}s")
         logger.debug(f"Found {len(plugin)} {plugin_type}(s) to run")
 
-        for p in plugin:
-            if self._stop.is_set():
-                logger.debug(f"Received a stop signal, skipping remaining {plugin_type}s")
-                return
-
+        interrupted_message = f"Received a stop signal, skipping remaining {plugin_type}s"
+        for p in interruptable_iter(plugin, self._stop, interrupted_message):
```

Comment:
Unrelated, but there's a typo: `plugins` not `plugin`

Reply:
Good catch. Fixed.